feat(checkout v3): Success page

isabellaenriquez opened the pull request:
Closes https://linear.app/getsentry/issue/BIL-1215/checkout-confirmation-screen

# immediate charge
<img width="1512" height="858" alt="Screenshot 2025-09-04 at 5 29 24 PM" src="https://github.com/user-attachments/assets/5ff85822-16ab-43d5-aeac-e4f8159f23ef" />

# smaller screen
<img width="870" height="860" alt="Screenshot 2025-09-04 at 5 29 34 PM" src="https://github.com/user-attachments/assets/9d774fa2-f1a5-47bd-8def-bf84cdaba860" />

# scheduled changes
<img width="1512" height="856" alt="Screenshot 2025-09-04 at 5 29 48 PM" src="https://github.com/user-attachments/assets/7c8f6818-811a-4f67-b528-62f4eac575bb" />

isabellaenriquez committed:
feat(checkout v3): Success page

isabellaenriquez committed:
some changes

isabellaenriquez committed:
receipt styling

isabellaenriquez committed:
animation

isabellaenriquez committed:
cleanup for transaction state

isabellaenriquez committed:
scheduled changes state

cobyeastwood183 committed:
feat(issue stream): use existing skeleton and keep loading per component (#97975)

Reused the loading skeleton and loading stream group components from
https://github.com/getsentry/sentry/pull/96854 but changed it back to
load per component again because loading all at once caused the state to
sometimes remain locked and fail tests.

isabellaenriquez committed:
Merge branch 'isabella/bil-1216' into isabella/bil-1215

isabellaenriquez committed:
cleanup + use creditApplied

thetruecpaul committed:
fix(typing): Remove sentry.integrations.slack.message_builder.notifications.issues from problem list (#98854)

Made minimal changes to allow mypy to pass with the file removed from
the problem list.

Normally, `ProjectNotification`s aren't guaranteed to have a `group`
property set. Hence the current in-code tendency towards accessing it
through `getattr` with a `None` default.

... but, in this case in particular, we're immediately passing it into
`SlackIssuesMessageBuilder::build`. That immediately accesses
`group.issue_category`, `group.organization`, etc — so it would already
throw if `group` was `None`.

Since we need `group` to be non-`None`, let's just take it directly.
(Which has the benefit of better typing!)

Test plan: `mypy` no errors; `pytest tests/sentry/integrations/slack/**`
no durable failures (one flaky failure that passed on first retry)

scttcper committed:
feat(issues): Convert similar stack trace to fc (#98835)

Removed the "visible" part of the state, I couldn't find anywhere still
setting visibility in the group merge store

kddubey committed:
ref(seer): Fixability only GPU (#98843)

[healthy](https://sentry.sentry.io/explore/traces/?aggregateField=%7B%22groupBy%22%3A%22sentry_region%22%7D&aggregateField=%7B%22yAxes%22%3A%5B%22count%28span.duration%29%22%5D%7D&aggregateField=%7B%22yAxes%22%3A%5B%22p50%28span.duration%29%22%5D%7D&aggregateField=%7B%22yAxes%22%3A%5B%22p99%28span.duration%29%22%5D%7D&field=id&field=span.op&field=span.description&field=span.duration&field=transaction&field=timestamp&field=fixability-model-device&field=sentry_region&interval=5m&mode=aggregate&project=6178942&query=span.description%3A%2Fv1%2Fautomation%2Fsummarize%2Ffixability%20fixability-model-device%3Acuda%3A0&statsPeriod=12h)

isabellaenriquez committed:
cleanup

evanpurkhiser committed:
feat(uptime): Add feature flag for detector IDs by default (#98849)

Adds organizations:uptime-detector-ids-by-default feature flag to
control whether uptime endpoints treat IDs as detector IDs by default,
instead of requiring the useDetectorId=1 query parameter.

When the feature flag is enabled, endpoints will use detector IDs by
default but still respect the useDetectorId query parameter for backward
compatibility. When disabled, endpoints preserve the legacy behavior of
treating IDs as ProjectUptimeSubscription IDs unless useDetectorId=1 is
explicitly provided.

This allows for gradual rollout of the detector ID change and quick
rollback if customer issues arise.

evanpurkhiser committed:
ref(uptime): Remove ProjectUptimeSubscriptionMode shim (#98821)

This isn't used in getsentry anymore

isabellaenriquez committed:
more cleanup + tests

volokluev committed:
Remove deprecated timestamp creation in indexer code (#98661)

lots of noise in the logs from this, update to the new convention

---------

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

malwilley committed:
feat(aci): Add check-ins table to cron monitor details page (#98834)

nora-shap committed:
chore(pr comment reactions): exempt task from sampling (#98859)

Temp change while we investigate this slow task

isabellaenriquez committed:
knip

Christinarlong committed:
fix(servicehook): Add RPC for bulk ServiceHook creation (#98662)

Commit:
fix(logs btn): remove extra margin (#98856)

**Before:**
<img width="154" height="92" alt="Screenshot 2025-09-04 at 1 38 47 PM"
src="https://github.com/user-attachments/assets/24a6fe5a-ad83-460d-b9fd-9bcee32db8d8"
/>

**After:**
<img width="166" height="107" alt="Screenshot 2025-09-04 at 1 38 51 PM"
src="https://github.com/user-attachments/assets/8adea54e-724e-4c9f-a9f7-5b4cd46c046f"
/>

natemoo-re committed:
fix(stories): hash scroll behavior (#98770)

Previously, hash scroll logic was inside of the mutation observer
leading to an infinite loop of `scrollIntoView()`.

To fix this, the scroll logic has been separated to a distinct hook and
is not tracked against a ref.

isabellaenriquez committed:
credit applied util

isabellaenriquez committed:
add comment

trevor-e committed:
fix(preprod): build info can be missing for processing builds (#98857)

These fields can be missing when a build hasn't processed yet so they
shouldn't be required.

evanpurkhiser committed:
feat(uptime): Always show links to uptime trace waterfall (#98839)

This effectively launches this feature.

isabellaenriquez committed:
fix type

aliu39 committed:
ref(replay): update summary endpoint blueprint and 404 for missing replays (#98764)

Closes [REPLAY-668: Update Sentry API
blueprint](https://linear.app/getsentry/issue/REPLAY-668/update-sentry-api-blueprint)
Closes [REPLAY-669: 404 if replay is missing from the date
range](https://linear.app/getsentry/issue/REPLAY-669/404-if-replay-is-missing-from-the-date-range)

---------

Co-authored-by: Michelle Zhang <56095982+michellewzhang@users.noreply.github.com>

isabellaenriquez committed:
handle NaN

vbro reviewed (approved):
Billing LGTM
Do you need Frontend or Design to review this PR?

trevor-e committed:
fix(preprod): make app fields nullable (#98858)

Similar to https://github.com/getsentry/sentry/pull/98857, makes these
API fields nullable.

calvin-codecov committed:
fix(codecov): Fix UI audit items for TA onboarding (#98782)

Closes
https://linear.app/getsentry/issue/CCMRG-1588/ta-onboarding-ui-audit-fixes

Going down the list above...
- removed "Prerequisites to connect" section from preonboarding
- set max-width for onboarding content to 1000px while outer box stays
at 1200px
- updated image at top of onboarding steps for both light and dark
- (Already done elsewhere): updated header color
- removed Done button and yellow "save your token before we hide it"
banner from token regeneration modal and onboarding generation step
- added right padding to the token title and token snippet boxes to make
more room for copy button
- changed GHActions workflow example from img to code snippet
- changed inline code blocks from inline snippets to pink text
- updated UI copy of second bullet of "View results and insights" step

iamrajjoshi committed:
:wrench: chore: update WorkflowEngine models in RuleSnooze Endpoint (#98865)

evanpurkhiser committed:
ref(uptime): Remove unused feature flags (#98867)

These were removed in a4ed9665db9ae7ecaaacf7eb674ab9f3a8933359

adrianviquez committed:
fix(prevent) attempt to fix redirection in prevent route (#98840)

This PR intends to address redirection issues w/ the /prevent/ route.

Currently, `.../prevent/`, `.../prevent/tests/` and
`.../prevent/tokens/` redirect to `.../codecov/`, breaking the UI. This
PR intends to amend this

Closes
https://linear.app/getsentry/issue/CCMRG-1508/fix-routing-when-explicitly-route-to-prevent-routes

aliu39 committed:
fix(replay): use IN condition for trace queries to avoid recursion depth (#98869)

Fixes
[SENTRY-4F08](https://sentry.sentry.io/issues/6856703798/events/21489a34b746431d84762b16f0e02d85/)
Closes [REPLAY-671: Recursion depth exceeded for trace
queries](https://linear.app/getsentry/issue/REPLAY-671/recursion-depth-exceeded-for-trace-queries)

joshuarli committed:
pre-commit: add missing hooks from the spec (#98863)

kddubey committed:
fix(bug-prediction): Check org when getting event (#98877)

defaulting to `None` disables this fetch until [the seer
PR](https://github.com/getsentry/seer/pull/3297) is deployed

roaga committed:
chore(explorer): Get more spans (#98873)

We have a lot of orphan nodes, so i think getting more complete traces
will help

JoshFerge committed:
refactor: Move datascrubbing module to relay package (#98850)

## Summary
- Move `src/sentry/datascrubbing.py` to
`src/sentry/relay/datascrubbing.py`
- Move test file to `tests/sentry/relay/test_datascrubbing.py`
- Update all import statements in affected files

## Rationale
The datascrubbing module is most closely coupled with relay
configuration and PII handling functionality. Moving it to the relay
package better organizes the codebase by grouping related functionality
together.

## Test plan
- [x] Moved test file maintains existing coverage
- [x] All import statements updated correctly
- [x] Run existing tests to ensure no regressions

part of
https://linear.app/getsentry/issue/RTC-1147/move-misc-top-level-modules-into-relevant-folders

vgrozdanic committed:
ref(ai): do not log model cost failures in dev mode (#98886)

obostjancic committed:
fix(agents-insights): explore links (#98797)

vgrozdanic committed:
feat(ai): Allow disabling model cost fetching task (#98888)

When running sentry as self hosted in air gapped environment, or with
strict egress firewall rules, model cost fetching will fail and cause a
lot of noise in the logs. Introducing this options gives users ability
to turn this feature off in those scenarios.

This should help with https://github.com/getsentry/sentry/issues/98338

closes [TET-1051: issue on self hosting sentry when fetching current
model prices from
openrouter](https://linear.app/getsentry/issue/TET-1051/issue-on-self-hosting-sentry-when-fetching-current-model-prices-from)

constantinius committed:
ref(analytics): introduce EventEnvelope class to store timestamp and UUID of an event to not clash with any fields (#98592)

Closes
https://linear.app/getsentry/issue/TET-809/analytics-investigate-potential-attribute-clash-with-baseinherited

obostjancic committed:
fix(ai-insights): fallback to gen_ai span count (#98891)

shellmayr committed:
feat(agent-monitoring): add onboarding for langchain and langgraph integrations (#98896)

- Add in-product onboarding for LangChain and LangGraph integrations
- Move verify code snippets to separate verify steps for all frameworks 
- Move onboarding to content blocks

armenzg committed:
feat(deletions): Add IssuePlatform batching back (#98815)

In #96795 we added a new task for deleting nodestore events separate
from the group deletion task due to its slowness.

Once it went live, we noticed that it would fail when retrieving the
Group from the group_ids passed. This was due to the parent task having
deleted the groups before the new task would be executed.

In order to prevent the problem, we temporarily removed the batching
support for Issue Platform in #96958 since there would not be customer
impact and TTL would eventually delete the data.

This change does the following:
* Calls the new task with the `group_ids` sorted
* Add a new parameter containing the associated `times_seen` for those
group ids

All of this batching is to ensure we don't call Snuba light deletions
for more than
[ISSUE_PLATFORM_MAX_ROWS_TO_DELETE](https://github.com/getsentry/snuba/blob/54feb15b7575142d4b3af7f50d2c2c865329f2db/snuba/datasets/configuration/issues/storages/search_issues.yaml#L139)
rows.

Fixes [ID-840: Add issue platform batching logic
back](https://linear.app/getsentry/issue/ID-840/add-issue-platform-batching-logic-back)

---------

Co-authored-by: Yuval Mandelboum <yuval.mandelboum@sentry.io>

armenzg committed:
chore(deletions): Remove sequential approach to deleting events (#98899)

Now that we delete events inside of the new task from #96795, we don't
need the old code anymore.

The original PR got reviewed in #98791

cmanallen committed:
ref(replays): Replace boolean flag with rollout percentage (#98800)

JonasBa commented on static/gsApp/views/amCheckout/checkoutSuccess.tsx:
```diff
@@ -0,0 +1,688 @@
+  }
+`;
+
+const Title = styled('h1')`
```
@isabellaenriquez could you take a look at heading, text and container components here? I think you could remove a lot of the custom CSS from this page if you used them

JonasBa committed:
layout: add properties to container (#97842)

We are seeing folks having to use `style` attribute on the layout
components. This is in part due to the unsupported properties (margins,
selective borders or flex item rules like grow, shrink or basis).

I'm adding the rules so that styled can be removed. I will periodically
monitor this - there are some cases where style was used for props like
height that are supported. Hopefully that was just a fluke, but it's
something to keep an eye on if it continues to happen, as we may look to
either disallow style from the component entirely or remove the
supported props from the style type.

---------

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

isabellaenriquez committed:
Merge branch 'master' into isabella/bil-1216

isabellaenriquez committed:
Merge branch 'isabella/bil-1216' into isabella/bil-1215

isabellaenriquez committed:
styling improvements
